llvm/test/CodeGen/WebAssembly/simd-relaxed-fma.ll

```diff
@@ -0,0 +1,249 @@
+; STRICT-NEXT:    return $pop1
+  %fma = tail call <4 x float> @llvm.fmuladd(<4 x float> %a, <4 x float> %b, <4 x float> %c)
+  ret <4 x float> %fma
+}
+
+define <4 x float> @fma_contract_4xf32(<4 x float> %a, <4 x float> %b, <4 x float> %c) {
+; RELAXED-LABEL: fma_contract_4xf32:
+; RELAXED:         .functype fma_contract_4xf32 (v128, v128, v128) -> (v128)
+; RELAXED-NEXT:  # %bb.0:
+; RELAXED-NEXT:    f32x4.extract_lane $push2=, $0, 0
+; RELAXED-NEXT:    f32x4.extract_lane $push1=, $1, 0
+; RELAXED-NEXT:    f32x4.extract_lane $push0=, $2, 0
+; RELAXED-NEXT:    call $push3=, fmaf, $pop2, $pop1, $pop0
+; RELAXED-NEXT:    f32x4.splat $push4=, $pop3
+; RELAXED-NEXT:    f32x4.extract_lane $push7=, $0, 1
+; RELAXED-NEXT:    f32x4.extract_lane $push6=, $1, 1
+; RELAXED-NEXT:    f32x4.extract_lane $push5=, $2, 1
+; RELAXED-NEXT:    call $push8=, fmaf, $pop7, $pop6, $pop5
+; RELAXED-NEXT:    f32x4.replace_lane $push9=, $pop4, 1, $pop8
+; RELAXED-NEXT:    f32x4.extract_lane $push12=, $0, 2
+; RELAXED-NEXT:    f32x4.extract_lane $push11=, $1, 2
+; RELAXED-NEXT:    f32x4.extract_lane $push10=, $2, 2
+; RELAXED-NEXT:    call $push13=, fmaf, $pop12, $pop11, $pop10
+; RELAXED-NEXT:    f32x4.replace_lane $push14=, $pop9, 2, $pop13
+; RELAXED-NEXT:    f32x4.extract_lane $push17=, $0, 3
+; RELAXED-NEXT:    f32x4.extract_lane $push16=, $1, 3
+; RELAXED-NEXT:    f32x4.extract_lane $push15=, $2, 3
+; RELAXED-NEXT:    call $push18=, fmaf, $pop17, $pop16, $pop15
+; RELAXED-NEXT:    f32x4.replace_lane $push19=, $pop14, 3, $pop18
+; RELAXED-NEXT:    return $pop19
+;
+; STRICT-LABEL: fma_contract_4xf32:
+; STRICT:         .functype fma_contract_4xf32 (v128, v128, v128) -> (v128)
+; STRICT-NEXT:  # %bb.0:
+; STRICT-NEXT:    f32x4.extract_lane $push2=, $0, 0
+; STRICT-NEXT:    f32x4.extract_lane $push1=, $1, 0
+; STRICT-NEXT:    f32x4.extract_lane $push0=, $2, 0
+; STRICT-NEXT:    call $push3=, fmaf, $pop2, $pop1, $pop0
+; STRICT-NEXT:    f32x4.splat $push4=, $pop3
+; STRICT-NEXT:    f32x4.extract_lane $push7=, $0, 1
+; STRICT-NEXT:    f32x4.extract_lane $push6=, $1, 1
+; STRICT-NEXT:    f32x4.extract_lane $push5=, $2, 1
+; STRICT-NEXT:    call $push8=, fmaf, $pop7, $pop6, $pop5
+; STRICT-NEXT:    f32x4.replace_lane $push9=, $pop4, 1, $pop8
+; STRICT-NEXT:    f32x4.extract_lane $push12=, $0, 2
+; STRICT-NEXT:    f32x4.extract_lane $push11=, $1, 2
+; STRICT-NEXT:    f32x4.extract_lane $push10=, $2, 2
+; STRICT-NEXT:    call $push13=, fmaf, $pop12, $pop11, $pop10
+; STRICT-NEXT:    f32x4.replace_lane $push14=, $pop9, 2, $pop13
+; STRICT-NEXT:    f32x4.extract_lane $push17=, $0, 3
+; STRICT-NEXT:    f32x4.extract_lane $push16=, $1, 3
+; STRICT-NEXT:    f32x4.extract_lane $push15=, $2, 3
+; STRICT-NEXT:    call $push18=, fmaf, $pop17, $pop16, $pop15
+; STRICT-NEXT:    f32x4.replace_lane $push19=, $pop14, 3, $pop18
+; STRICT-NEXT:    return $pop19
+  %fma = tail call contract <4 x float> @llvm.fma(<4 x float> %a, <4 x float> %b, <4 x float> %c)
```

Comment:
No need for the contract flag here since by definition fma happens in one rounding step

```suggestion
  %fma = call <4 x float> @llvm.fma(<4 x float> %a, <4 x float> %b, <4 x float> %c)
```